core/tests/protractor_desktop/contributorDashboard.js

```diff
@@ -114,13 +114,34 @@ describe('Contributor dashboard page', function() {
   });
 
   it('should allow user to switch to translate text tab', async function() {
+    await users.login(USER_EMAILS[0]);
     await contributorDashboardPage.get();
     await contributorDashboardPage.navigateToTranslateTextTab();
     await contributorDashboardTranslateTextTab.changeLanguage(GERMAN_LANGUAGE);
     await contributorDashboardTranslateTextTab.expectSelectedLanguageToBe(
       GERMAN_LANGUAGE);
+    await users.logout();
```

Comment:
Why were these changes necessary here?

Reply:
Because if we try to change the language without login, we will encounter a console log error and this test would fail. We make a post request whenever a user changes a language from the dropdown (which will return an error if the user is not logged in).

Reply:
But this seems odd. We should make sure that the behaviour works correctly for non-logged-in users, too. If it wasn't throwing an error before, it should not throw one now.

Reply:
I have removed this now. I have added a check in the front-end itself to make a post request for saving the language preference only when user is logged-in.